bond/bond_test.go

```diff
@@ -38,110 +40,95 @@ const (
 			"cniVersion": "%s",
 			"mode": "%s",
 			"failOverMac": 1,
-			"linksInContainer": true,
+			"linksInContainer": %s,
 			"miimon": "100",
-			"mtu": 1400,
+			"mtu": %s,
 			"links": [
 				{"name": "net1"},
 				{"name": "net2"}
 			]
 		}`
 	ActiveBackupMode = "active-backup"
 	BalanceTlbMode   = "balance-tlb"
+	DefaultMTU       = 1400
 )
 
 var Slaves = []string{Slave1, Slave2}
 
 var _ = Describe("bond plugin", func() {
 	var podNS ns.NetNS
+	var initNS ns.NetNS
+	var linkAttrs []netlink.LinkAttrs
+	var linksInContainer bool
+
+	When("links are in container`s network namespace at initial state (meaning linksInContainer is true)", func() {
+		BeforeEach(func() {
+			var err error
+			linksInContainer = true
+			linkAttrs = []netlink.LinkAttrs{
+				{Name: Slave1},
+				{Name: Slave2},
+			}
+			podNS, err = testutils.NewNS()
+			Expect(err).NotTo(HaveOccurred())
+			addLinksInNS(podNS, linkAttrs)
+		})
 
-	BeforeEach(func() {
-		var err error
-		podNS, err = testutils.NewNS()
-		Expect(err).NotTo(HaveOccurred())
-
-		for _, ifName := range Slaves {
-			err = podNS.Do(func(ns.NetNS) error {
-				defer GinkgoRecover()
-				err = netlink.LinkAdd(&netlink.Dummy{
-					LinkAttrs: netlink.LinkAttrs{
-						Name: ifName,
-					},
-				})
-				Expect(err).NotTo(HaveOccurred())
-				_, err := netlink.LinkByName(ifName)
-				Expect(err).NotTo(HaveOccurred())
-				return nil
-			})
-		}
-		Expect(err).NotTo(HaveOccurred())
-	})
-
-	AfterEach(func() {
-		Expect(podNS.Close()).To(Succeed())
-		Expect(testutils.UnmountNS(podNS)).To(Succeed())
-	})
-
-	It("verifies a plugin is added and deleted correctly", func() {
-		args := &skel.CmdArgs{
-			ContainerID: "dummy",
-			Netns:       podNS.Path(),
-			IfName:      IfName,
-			StdinData:   []byte(fmt.Sprintf(Config, "0.3.1", ActiveBackupMode)),
-		}
-
-		err := podNS.Do(func(ns.NetNS) error {
-			defer GinkgoRecover()
+		AfterEach(func() {
+			Expect(podNS.Close()).To(Succeed())
+			Expect(testutils.UnmountNS(podNS)).To(Succeed())
+		})
+		It("verifies a plugin is added and deleted correctly", func() {
+			args := &skel.CmdArgs{
```

Comment:
let's move args inside the before each. It's fine to test only with 1.0.0 version. We have other test to verify other versions.

Reply:
sounds good